synapse/handlers/cas_handler.py

```diff
@@ -127,78 +147,69 @@ def _parse_cas_response(
             )
         return user, attributes
 
-    async def _on_successful_auth(
+    def get_redirect_url(self, service_args: Dict[str, str]) -> str:
+        """
+        Generates a URL to the CAS server where the client should be redirected.
+
+        Args:
+            service_args: Additional arguments to include in the final redirect URL.
+
+        Returns:
+            The URL to redirect the client to.
+        """
+        args = urllib.parse.urlencode(
+            {"service": self._build_service_param(service_args)}
+        )
+
+        return "%s/login?%s" % (self._cas_server_url, args)
+
+    async def handle_ticket(
         self,
-        username: str,
         request: SynapseRequest,
-        client_redirect_url: str,
-        user_display_name: Optional[str] = None,
+        ticket: str,
+        client_redirect_url: Optional[str],
+        session: Optional[str],
     ) -> None:
-        """Called once the user has successfully authenticated with the SSO.
+        """
+        Called once the user has successfully authenticated with the SSO,
+        validates a CAS ticket sent by the client and completes the login process.
 
         Registers the user if necessary, and then returns a redirect (with
         a login token) to the client.
```

Comment:
this is only true if session_id is None. We should say so, and also say what we do if it is not.

Reply:
Ah, I failed to update this comment for sharing the endpoint. Good call, thanks!